lib/wificaptive/portal/index.html

```diff
@@ -338,12 +338,35 @@
       hideAlert();
       var ssid = document.getElementById('ssid').value;
       var pswd = document.getElementById('password').value;
+      var server = document.getElementById('server').value;
       if (!ssid) {
         displayAlert('Please enter your Wi-Fi name', "warning");
         return;
       }
 
-      var body = { ssid: ssid, pswd: pswd };
+      // if custom server form-group is visible, validate the url
+      let customServerVisible = document.getElementById('custom_server_form_group').style.display != 'none';
+      if (customServerVisible && server) {
+        try {
+          new URL(server);
+          // does it start with http:// or https://?
+          if (!server.startsWith('http://') && !server.startsWith('https://')) {
+            throw new Error();
+          }
+        } catch (_) {
+          displayAlert('Please enter a valid URL for your custom server', "warning");
+          return;
+        }
+      }
+
+      // strip trailing slashes
+      server = server.replace(/\/+$/, '');
+      if (!server) {
+        server = "https://trmnl.app";
```

Comment:
possible to leverage `API_BASE_URL` here instead of duplicating? not a deal breaker, but we may have to start rotating alias domains in the future so would like to keep as DRY as possible.